tensorflow/python/estimator/inputs/numpy_io.py

```diff
@@ -132,6 +132,8 @@ def input_fn():
     features = dict(zip(ordered_dict_x.keys(), features))
     if y is not None:
       target = features.pop(unique_target_key)
+      # undo modification to dictionary
+      del x[unique_target_key]
```

Comment:
Do you really need this line? The modified dict is the ordered_dict_x (a shadow copy of x). Input x is not modified.